src/test/groovy/io/openliberty/tools/gradle/GenerateFeaturesTest.groovy

```diff
@@ -85,6 +89,20 @@ class GenerateFeaturesTest extends AbstractIntegrationTest {
         assertEquals(expectedFeatures, features);
     }
 
+    @Test
+    public void noClassFiles() throws Exception {
+        // do not compile before running generateFeatures
+        runProcess(" generateFeatures");
+
+        // verify that generated features file was not created
+        assertFalse(newFeatureFile.exists());
+
+        // verify class files not found warning message
+        assertTrue(processOutput.contains("Could not find class files to generate features against. "
+                + "Liberty features will not be generated. Ensure your project has first been compiled."));
```

Comment:
We should make this warning a constant so we are not checking a String that may change. Probably best to have it in ci.common to use across Maven and Gradle as well.

Reply:
The message varies very slightly in wording between Maven and Gradle (class directories vs class files) so I don't think it should go in ci.common, but I created a constant in GenerateFeaturesTask